food-truck-api/src/main/java/food/truck/api/endpoint/TruckEndpoint.java

```diff
@@ -214,9 +239,9 @@ public Route createTruckRoute(@AuthenticationPrincipal User user, @PathVariable
         Long routeLocationId;
         long routeId;
         @NonNull
-        Instant arrivalTime;
+        LocalDateTime arrivalTime;
         @NonNull
-        Instant exitTime;
+        LocalDateTime exitTime;
```

Comment:
I think we should keep it as Instant, because an Instant has a time zone associated with it. That way, we can convert the time to UTC and store it; then, when we give the time to the frontend, the frontend knows it is a UTC time, and can convert it to the user's time zone. If we just give the backend a LocalDateTime and do .toLocalTime(), then people in two different time zones can put in the same "real" time, but the DB will only care about the "clock" time which will be different.

Reply:
The issue arises when UTC tries to correct for time zone. Since it's considering everything within a 24 hour day (since attaching day information causes problems like a route invalidating itself within 24 hours), it can destroy our time calculation by rolling one across midnight but not the other (see if you have an early-morning truck; the start rolls back across midnight for UTC, the end doesn't, now `isBefore` and `isAfter` don't give the correct answer). I guess we could add some custom logic to handle it, though.